app/main.py

```diff
@@ -1,34 +1,92 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.cell = (row, column)
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.decks = []
+        if start[0] == end[0]:
+            for i in range(start[1], end[1] + 1):
+                self.decks.append(Deck(start[0], i))
+        else:
+            for i in range(start[0], end[0] + 1):
+                self.decks.append(Deck(i, end[1]))
+        self.is_drowned = is_drowned
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> bool:
+        for deck in self.decks:
+            return (row, column) in deck.cell
+
+    def fire(self, row: int, column: int) -> None:
+        for deck in self.decks:
+            print(deck.cell)
+        if (row, column) in self.decks:
+            if not self.decks:
+                self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+
+    def __init__(self, ships: list) -> None:
+        self.field = {}
+        self.battle_field = []
+        for _ in range(10):
+            self.battle_field.append(["~" for _ in range(10)])
```

Comment:
It is redundant and better to do it in the loop due to the current ships and decks
Please Simplify code

Reply:
Didn't catch the idea.. Where do I need to create battle field? In Ship class?

Reply:
In which do I need to implement this method? 
I write new print func inside Battleship class. Did I write it correctly? Do I need to move this code to __str__ method?